server/handlers/component_handler.go

```diff
@@ -1367,3 +1374,173 @@ func prettifyCompDefSchema(entities []entity.Entity) []v1beta1.ComponentDefiniti
+			return
+		}
+		rw.WriteHeader(http.StatusOK)
+		_, _ = rw.Write([]byte("OCI image processed successfully"))
+		return
+	}
+
+	dirPath := r.FormValue("dir")
+	if dirPath != "" {
+		err := processDirectory([]byte(dirPath), h)
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("Error processing directory: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		rw.WriteHeader(http.StatusOK)
+		_, _ = rw.Write([]byte("Directory processed successfully"))
+		return
+	}
+
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		http.Error(rw, "Error retrieving file", http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+
+	fileContent, err := io.ReadAll(file)
+	if err != nil {
+		http.Error(rw, "Error reading file", http.StatusInternalServerError)
+		return
+	}
+
+	err = processFileContent(fileContent, h)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("Error processing file: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	_, _ = rw.Write([]byte("File uploaded and processed successfully"))
+}
+
+func processDirectory(data []byte, h *Handler) error {
+	tempFile, err := os.CreateTemp("", "upload-*.tar.gz")
+	if err != nil {
+		return fmt.Errorf("Error creating temporary file: %v", err)
+	}
+	defer os.Remove(tempFile.Name())
+
+	_, err = tempFile.Write(data)
+	if err != nil {
+		return fmt.Errorf("Error writing to temporary file: %v", err)
+	}
+
+	return processUploadedFile(tempFile.Name(), h)
+}
+
+func processUploadedFile(filePath string, h *Handler) error {
+	tempDir, err := os.MkdirTemp("", "extracted-")
+	if err != nil {
+		return fmt.Errorf("Error creating temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	err = extractFile(filePath, tempDir)
+	if err != nil {
+		return fmt.Errorf("Error extracting file: %v", err)
+	}
+	err = filepath.Walk(tempDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			if meshkitutils.IsYaml(path) {
+				fmt.Println(path)
+				content, err := os.ReadFile(path)
+				if err != nil {
+					return err
+				}
+				return processFileContent(content, h)
+			}
+			if meshkitutils.IsTarGz(path) || meshkitutils.IsZip(path) {
+				return processUploadedFile(path, h)
+			}
+		}
+		return nil
+	})
+
+	return err
+}
+
+func extractFile(filePath string, destDir string) error {
+	if meshkitutils.IsTarGz(filePath) {
+		return meshkitutils.ExtractTarGz(destDir, filePath)
+	} else if meshkitutils.IsZip(filePath) {
+		return meshkitutils.ExtractZip(destDir, filePath)
+	}
+	return fmt.Errorf("unsupported file type for extraction: %s", filePath)
+}
+
+func processFileContent(content []byte, h *Handler) error {
+	var tempMap map[string]interface{}
+	if err := json.Unmarshal(content, &tempMap); err != nil {
+		return fmt.Errorf("error unmarshalling JSON content: %w", err)
+	}
+
+	if schemaVersion, ok := tempMap["schemaVersion"].(string); ok {
+		switch schemaVersion {
```

Comment:
@Jougan-0 Adding so that it doesn't get missed.
Please use the approach as discussed as discussed on the call.

This serves the purpose but is not ideal.